From 58011bd9ef1097ccb6f16f5008147963c4c9d05a Mon Sep 17 00:00:00 2001
From: Chao Sun <sunchao@apache.org>
Date: Mon, 6 Jul 2015 16:17:21 -0700
Subject: [PATCH 0180/1164] Revert "HIVE-10573 - Improve Hive service for
 connection (Chao, reviewed by Thejas)"

Revert this commit since it doesn't include unit tests - replaced with HIVE-9934 + HIVE-10572

This reverts commit 6bc4cb53312ca4fc2036aeb78eab5fd207718087.
---
 .../auth/LdapAuthenticationProviderImpl.java       |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/service/src/java/org/apache/hive/service/auth/LdapAuthenticationProviderImpl.java b/service/src/java/org/apache/hive/service/auth/LdapAuthenticationProviderImpl.java
index 0c7cede..a095006 100644
--- a/service/src/java/org/apache/hive/service/auth/LdapAuthenticationProviderImpl.java
+++ b/service/src/java/org/apache/hive/service/auth/LdapAuthenticationProviderImpl.java
@@ -132,12 +132,14 @@ public void Authenticate(String user, String password) throws AuthenticationExce
       user  = user + "@" + ldapDomain;
     }
 
-    if (password == null || password.isEmpty() || password.getBytes()[0] == 0) {
-      throw new AuthenticationException("Error validating LDAP user:" +
-          " a null or blank password has been provided");
+    // setup the security principal
+    String bindDN;
+    if (baseDN == null) {
+      bindDN = user;
+    } else {
+      bindDN = "uid=" + user + "," + baseDN;
     }
 
-    // user being authenticated becomes the bindDN and baseDN or userDN is used to search
     env.put(Context.SECURITY_AUTHENTICATION, "simple");
     env.put(Context.SECURITY_PRINCIPAL, user);
     env.put(Context.SECURITY_CREDENTIALS, password);
-- 
1.7.9.5

